A guide / short intro for ITS Logo versions #1496

Alexey-Slyusar opened the pull request:
Lars, Eric, please take a look at the Logo guide. It is very possible that the document overloads a reader with too much additional info and has some unclear parts. But I tried in this document to introduce the reader with great MIT Logo Group intellectual heritage that we have collected here in some depth.

Alexey-Slyusar commented:
I've squashed commits twice. But it seems that it is needed to do only once - right before PR.

Alexey-Slyusar commented:
I want to add at least one more thing to the document. Although I am not going to change the existing parts of the document.

I'll add the additional part in the morning (US West Coast evening).

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+Brian Harvey
+
+Logo was initially created by Wally Feurzeig, Seymour Papert, and Cynthia
+Solomon in 1967 as part of National Science Foundation sponsored research
```
"as part of **a** National ... project"

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+in the Artificial Intelligence Lab and the Division for Study and Research in
+Education, using large research computer systems, such as ITS powered PDP-10.
+
+Our goal is to make that early Logo systems available to wider auditory of
```
"to **a** wider auditory"

Maybe audience is a better word than auditory here?

Alexey-Slyusar replied:
change for audience.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+
+BBN PDP-10 LOGO system was implemented originally by Walter B. Weiner with
+the assistance of Paul Wexelblat and Charles R. Morgan at BBN. 
+We have source file with a copyright from 1970. It is version modified at National
```
"**The** BBN PDP-10 system..."  
"We have **the** source file*  
"It is a version modified at..."

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+Research Council, Ottawa, Canada by A.G. Smith and R.A.Orchard.
+That version of BBN Logo for PDP-10 TOPS-10 and TENEX was assembled with MACRO-10
+in ITS, and dumped with the DECUUO bootstrap to get ITS executable binary by Lars
+Brinkhoff. 
```
"to get **an** ITS executable binary."

I think it's better to remove my name and focus on the historical events.

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+
+Here is some BBN Logo related documentation:
+
+ Bolt Beranek and Newman, Inc., Report no. 2165 vol. 4 "The Logo Processor: A Gide 
```
"G**u**uide"

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+and MIDAS and enhanced with turtle graphics and other capabilities at MIT.
+
+We have pretty stable CLOGO version 49 binary as well as good-commented source code for
+the version.
```
"We have **a** pretty stable..."  
"*well*-commented source code"

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+### MIT LISP LOGO
+
+LISP LOGO is an implementation of the LOGO language written in MACLISP for the ITS, TEN50
+and TENEX PDP-10 systems, and MULTICS. The system were implemented by Ira Goldstein,
```
"The system **was** implemented"

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+Henry Lieberman in the early 70's at MIT.
+One of the reason for this implementation was to provide a natural transition to the more
+powerful computational world of LISP as the Logo programmer grows more sophisticated.
+When desired, the one has access to all of the capabilities of MACLISP.
```
"When desired, one has access"

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+- MUSIC BOX - the package provides Logo music capabilities.
+
+The LLOGO variant of LISP LOGO was recently brought back to life during several amazing
+hacking session performed by Eric Swenson and Lars Brinkhoff. As a result, we
```
No need to praise Swenson and Brinkhoff here.

Alexey-Slyusar replied:
I do not quite agree, but I obey.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+[MIT A.I. Memo 307A "LLOGO: An Implementation of LOGO in LISP"](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwjNkYz73JngAhXGkywKHRfMDuQQFjAAegQICRAC&url=https%3A%2F%2Fdspace.mit.edu%2Fbitstream%2Fhandle%2F1721.1%2F6221%2FAIM-307a.pdf&usg=AOvVaw1iwaMXNU6NgcYAFToTxKlY)
+
+[MIT A.I. Memo 361 "The TV Turtle. A Logo Graphics System for Raster Displays"](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=2&cad=rja&uact=8&ved=2ahUKEwj05u7d35ngAhVNl4sKHb2vAIQQFjABegQICRAC&url=https%3A%2F%2Fdspace.mit.edu%2Fbitstream%2Fhandle%2F1721.1%2F5773%2FAIM-361.pdf%3Fsequence%3D2&usg=AOvVaw3reoCDZkG-ZAyhdEe2B4G_)
+
```
These two links go through Google.  Please make the direct link to dspace.mit.edu.

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+In accordance with Leigh L Klotz Jr. talk at comp.lang.logo channel 
+"around 1977-1978, Gary Drescher and someone else whose name escapes me at the moment
+wrote a version of Logo in Pascal, as part of a project with Texas Instruments, for
+the TI 99/4 Home Computer, because Pascal was thefonly high-level language supported.
```
"Pascal was **the only**"

Alexey-Slyusar replied:
done.

larsbrinkhoff commented on doc/logo.md:
```diff
@@ -0,0 +1,177 @@
+MIT APLOGO was developed by Stephen Hain, Patrick G. Sobalvarro and Leigh L Klotz Jr.
+It was developed and cross-compilled for the Apple-II-Plus Personal Microcomputer on
+PDP-10 at the MIT LOGO Group. It is direct predecessor for Terrapin Logo.
+We have source code for assembling Preliminary Version from 8/06/80 at `its/src/aplogo`
```
APLOGO has been updated to an improved 1981 version.

Alexey-Slyusar replied:
done.

larsbrinkhoff commented:
Thank you.  There's a wealth of information here, which I think is good.

I made some minor comments.

larsbrinkhoff commented:
Hello @leighklotz,

Sending this message just to keep you in the loop regarding Logo restoration here.  @Alexey-Slyusar is adding a document to summarize the implementations available.

Alexey-Slyusar commented:
Lars, I've added a section in the latest commit by `git commit --amend -C HEAD` and did `git push -f`.
Please take a look at this new section at the end of the document too.
And thank you very much for the comments. I am working on it.

I've added  @leighklotz as a reviewer for the document. It will be great if @leighklotz finding some time to take a look at the document. Thanks in advance!

And yes, thank you very much for fearless-git project. It is very helpful. But I still not sure that I do right things with git actually :-)

Alexey-Slyusar commented:
Oh, it seems that I have not had added @leighklotz as a reviewer actually. I can add only PDP-10 / ITS group members. @larsbrinkhoff I think it will be great to make invite for @leighklotz!

larsbrinkhoff commented:
Alexey, I believe Mr Klotz will get a message from GitHub when his user name is mentioned with an @.

Alexey-Slyusar commented:
I see, Lars, thank you!

Alexey-Slyusar commented:
I learned that Eric didn't have a chance to take a look at the document. I
believe that it is right idea to wait with the merging until Eric have some
free time to review. We are not in hurry and the quality of the document
benefit a lot from Eric's review.

вс, 17 февр. 2019 г. в 13:40, Lars Brinkhoff <notifications@github.com>:

> Alexey, I believe Mr Klotz will get a message from GitHub when his user
> name is mentioned with an @.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PDP-10/its/pull/1537#issuecomment-464440651>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AoaupCbK8XUY3JftdqG02ETtf4A0b0dqks5vOTGogaJpZM4a_HO->
> .
>

eswenson1 commented:
@Alexey-Slyusar No, it is not necessary to wait. Feel free to merge. One reviewer is good enough. If I find something I don't like about it, I'll enter a ticket later. Lars approved your PR, so it is good to go.

larsbrinkhoff commented:
It's not approved yet.

Can you please squash everything together into one single commit and push that?

larsbrinkhoff commented:
Sorry, I see you did that already.  Can you just update the commit message?

Alexey-Slyusar commented:
Participating a morning meeting at work. So my office duties swallowed me
up entierly for a while.

пн, 18 февр. 2019 г. в 10:33, Lars Brinkhoff <notifications@github.com>:

> Sorry, I see you did that already. Can you just update the commit message?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PDP-10/its/pull/1537#issuecomment-464617888>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AoaupAJrvf0tIfIivJVeZ4tzvDsXyOR5ks5vOldKgaJpZM4a_HO->
> .
>

Alexey-Slyusar commented:
Lars, I've updated the commit message.

larsbrinkhoff commented:
I think the title of this pull request is almost a good commit message.  I updated it and approved.  Alexey, you click the merge button if you think it's ready.

Alexey-Slyusar commented:
Lars, thank you!!!
